operation/fhir-operation-everything/src/main/java/com/ibm/fhir/operation/everything/EverythingOperation.java

```diff
@@ -0,0 +1,329 @@
+    protected MultivaluedMap<String, String> parseQueryParameters(Parameters parameters) {
+        MultivaluedMap<String, String> queryParameters = new MultivaluedHashMap<String, String>();
+        Parameter countParameter = getParameter(parameters, SearchConstants.COUNT);
+        if (countParameter == null) {
+            queryParameters.add(SearchConstants.COUNT, SearchConstants.MAX_PAGE_SIZE + "");
+        } else {
+            queryParameters.add(SearchConstants.COUNT, countParameter.getValue().as(FHIR_INTEGER).getValue() + "");
+        }
+        // We use gt/lt here in an effort to be more liberal in terms of what we return
+        // https://ibm-watsonhealth.slack.com/archives/C14JTTR6C/p1614181836050500?thread_ts=1614180853.047300&cid=C14JTTR6C
+        Parameter startParameter = getParameter(parameters, START_QUERY_PARAMETER);
+        if (startParameter != null) {
+            queryParameters.add(DATE_QUERY_PARAMETER, STARTING_FROM + startParameter.getValue().as(FHIR_DATE).getValue());
+        }
+        Parameter endParameter = getParameter(parameters, END_QUERY_PARAMETER);
+        if (endParameter != null) {
+            queryParameters.add(DATE_QUERY_PARAMETER, UP_UNTIL + endParameter.getValue().as(FHIR_DATE).getValue());
+        }
+        Parameter sinceParameter = getParameter(parameters, SINCE_QUERY_PARAMETER);
+        if (sinceParameter != null) {
+            queryParameters.add(LAST_UPDATED_QUERY_PARAMETER, STARTING_FROM + sinceParameter.getValue().as(FHIR_INSTANT).getValue());
+        }
+        return queryParameters;
+    }
+
+    /**
+     * @param parameters the {@link Parameters} object
+     * @return the list of patient subresources that will be included in the $everything operation, as provided by the user
+     * @throws FHIRSearchException 
+     */
+    protected List<String> getOverridenIncludedResourceTypes(Parameters parameters) throws FHIRSearchException {
+        List<String> typeOverrides = new ArrayList<>();
+        Parameter typesParameter = getParameter(parameters, SearchConstants.RESOURCE_TYPE);
+        if (typesParameter == null) {
+            return typeOverrides;
+        }
+        String typeOverridesParam = typesParameter.getValue().as(FHIR_STRING).getValue();
+        String[] typeOverridesList = typeOverridesParam.split(SearchConstants.JOIN_STR);
+        List<String> unknownTypes= new ArrayList<>();
+        for (String typeOverride : typeOverridesList) {
+            if (!defaultResourceTypes.contains(typeOverride)) {
+                unknownTypes.add(typeOverride);
+            } else {
+                typeOverrides.add(typeOverride.trim());
+            }
+        }
+        FHIRRequestContext requestContext = FHIRRequestContext.get();
+        if (!HTTPHandlingPreference.LENIENT.equals(requestContext.getHandlingPreference()) && !unknownTypes.isEmpty()) {
+            throw new FHIRSearchException("The following types are not known: " + String.join(",", unknownTypes));
```

Comment:
The types might be known, just not supported for patient compartment. Might be better to say "The following resource types are not supported by this operation: ". In the exception, should set an issue type of INVALID - see SearchExceptionUtil.buildInvalidException() - this will translate to a 4xx HTTP code rather than a 500 code. Also, if in lenient mode, we should log the message.

Reply:
Done.